test/OpenTelemetry.Tests/Trace/TracerProviderSdkTest.cs

```diff
@@ -263,38 +271,301 @@ public void TracerProvideSdkCreatesActivitySource()
             testActivityProcessor.StartAction =
                 (a) =>
                 {
+                    Assert.False(Sdk.SuppressInstrumentation);
+                    Assert.True(a.IsAllDataRequested); // If Proccessor.OnStart is called, activity's IsAllDataRequested is set to true
                     startCalled = true;
                 };
 
             testActivityProcessor.EndAction =
                 (a) =>
                 {
+                    Assert.False(Sdk.SuppressInstrumentation);
+                    Assert.True(a.IsAllDataRequested); // If Processor.OnEnd is called, activity's IsAllDataRequested is set to true
                     endCalled = true;
                 };
 
-            TestInstrumentation testInstrumentation = null;
+            var emptyActivitySource = new ActivitySource(string.Empty);
+            Assert.False(emptyActivitySource.HasListeners()); // No ActivityListener for empty ActivitySource added yet
+
+            // No AddLegacyOperationName chained to TracerProviderBuilder
             using var tracerProvider = Sdk.CreateTracerProviderBuilder()
                         .AddProcessor(testActivityProcessor)
-                        .AddDiagnosticSourceInstrumentation((adapter) =>
-                        {
-                            testInstrumentation = new TestInstrumentation(adapter);
-                            return testInstrumentation;
-                        })
                         .Build();
 
-            var adapter = testInstrumentation.Adapter;
-            Activity activity = new Activity("test");
+            Assert.False(emptyActivitySource.HasListeners()); // No listener for empty ActivitySource even after build
+
+            Activity activity = new Activity("Test");
+            activity.Start();
+            activity.Stop();
+
+            Assert.False(startCalled); // Processor.OnStart is not called since we did not add any legacy OperationName
+            Assert.False(endCalled); // Processor.OnEnd is not called since we did not add any legacy OperationName
+        }
+
+        // Test to check that TracerProvider samples a legacy activity using a custom Sampler and calls Processor.OnStart and Processor.OnEnd for the
+        // legacy activity when the correct legacy OperationName is provided to TracerProviderBuilder.
+        [Fact]
+        public void SdkSamplesAndProcessesLegacyActivityWithRightConfig()
+        {
+            bool samplerCalled = false;
```

Comment:
this is unused?

Reply:
There is an Assert statement for it in the beginning of `StartAction` in the `TestActivityProcessor`.